src/core/transpile.jl

```diff
@@ -2064,3 +2101,68 @@ function transpile(
 
     return circuit
 end
+
+struct DecomposeControlledGatesTranspiler <: Transpiler end
```

Comment:
Perhaps we can rename this to `DecomposeSingleTargetSingleControlGatesTranspiler`, and then let everything pass through that is not that. Then, the `UnsupportedGatesTranspiler` can catch anything that wasn't transpiled. This way we are not as dependent on the ordering of transpilers. What do you think?

Reply:
yes thats is better, updated